src/auto-evo/Miche.cs

```diff
@@ -0,0 +1,212 @@
+﻿namespace AutoEvo;
+
+using System.Collections.Generic;
+using System.Linq;
+
+public class Miche
```

Comment:
This seems like a core class of the miche feature, right? So I think this class really ought to have a summary comment. Also I see quite a lot of memory allocations here. For example `GetLeafNodes` should probably be written to take in a list to put the results in to start guiding the code here towards a way of writing it that minimizes memory allocations where it is easy to do (I'm not going to ask to remove all repeated memory allocations but where I think they are easy to avoid it would be worth it in order to ensure auto-evo doesn't add a lot of garbage collector pressure while running).

Reply:
I'm not sure if "A node for the Miche Tree" is enough, but it seems to follow the general format of the others summaries.

Reply:
That would work as a summary, but in addition I'd like to see a `remarks` section either explaining the miche tree shortly or linking to the miche tree class and then that class having an overall explanation for what the miche tree does.